kt-android-runtime/runtime/src/main/java/io/sdkgen/runtime/SdkgenHttpClient.kt

```diff
@@ -0,0 +1,229 @@
+        }
+        if (variant.isNotEmpty()) {
+            bcp47Tag.append(sep).append(variant)
+        }
+
+        return bcp47Tag.toString()
+    }
+
+    @SuppressLint("HardwareIds")
+    private fun makeDeviceObj(): JsonObject {
+        return JsonObject().apply {
+            addProperty("type", "android")
+            addProperty("fingerprint", Settings.Secure.getString(applicationContext.contentResolver, Settings.Secure.ANDROID_ID))
+            add("platform", JsonObject().apply {
+                addProperty("version", Build.VERSION.RELEASE)
+                addProperty("sdkVersion", Build.VERSION.SDK_INT)
+                addProperty("brand", Build.BRAND)
+                addProperty("model", Build.MODEL)
+            })
+
+            try {
+                addProperty("version", applicationContext.packageManager.getPackageInfo(applicationContext.packageName, 0).versionName)
+            } catch (e: Exception) {
+                addProperty("version", "unknown")
+            }
+
+            addProperty("language", language())
+            add("screen", JsonObject().apply {
+                val manager = applicationContext.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+                val display = manager.defaultDisplay
+                val size = Point()
+                display.getSize(size)
+                addProperty("width", size.x)
+                addProperty("height", size.y)
+            })
+
+            val pref = applicationContext.getSharedPreferences("api", Context.MODE_PRIVATE)
+            if (pref.contains("deviceId"))
+                addProperty("id", pref.getString("deviceId", null))
+        }
+    }
+
+    suspend fun makeRequest(functionName: String, bodyArgs: JsonObject?, timeoutMillis: Long = defaultTimeoutMillis): InternalResposne = suspendCoroutine { continuation ->
+        val httpTimer = Timer()
+        try {
+            val body = JsonObject().apply {
+                addProperty("id", callId())
+                add("device", makeDeviceObj())
+                addProperty("name", functionName)
+                add("args", bodyArgs ?: JsonObject())
+            }
+
+            val request = Request.Builder()
+                .url("https://$baseUrl/$functionName")
+                .post(body.toString().toRequestBody("application/json; charset=utf-8".toMediaType()))
+                .build()
+
+            val call = httpClient.newCall(request)
+            try {
+                httpTimer.schedule(timerTask { call.timeout() }, timeoutMillis)
+                val response = call.execute()
+                httpTimer.cancel()
+                if (response.code == 502) {
+                    continuation.resume(
+                        InternalResposne(
+                            JsonObject().apply {
+                                addProperty("type", "Fatal")
+                                addProperty("message", "Erro Fatal (502) - Tente novamente")
```

Comment:
Let's avoid hardcoded portuguese strings

Reply:
Yes, I've commented with @dgadelha that I still intend to migrate the hardcoded strings to android's R class and translate to english and spanish